runtime/gc_base/gc_internal.h

```diff
@@ -193,10 +193,10 @@ extern J9_CFUNC UDATA j9gc_get_overflow_safe_alloc_size(J9JavaVM *javaVM);
 extern J9_CFUNC IDATA j9gc_objaccess_mixedObjectReadI32(J9VMThread *vmThread, j9object_t srcObject, UDATA offset, UDATA isVolatile);
 extern J9_CFUNC j9object_t J9AllocateObject(J9VMThread *vmContext, J9Class *clazz, UDATA allocateFlags);
 extern J9_CFUNC UDATA j9gc_ext_is_marked(J9JavaVM *javaVM, j9object_t objectPtr);
-extern J9_CFUNC void J9WriteBarrierPreStore(J9VMThread *vmThread, J9Object *dstObject, fj9object_t *dstAddress, J9Object *srcObject);
-extern J9_CFUNC void J9WriteBarrierJ9PreClassStore(J9VMThread *vmThread, J9Object *dstObject, J9Object **dstAddress, J9Object *srcObject);
+extern J9_CFUNC void J9WriteBarrierPre(J9VMThread *vmThread, J9Object *dstObject, fj9object_t *dstAddress, J9Object *srcObject);
```

Comment:
It's unfortunated that Pre variant is inconsistent in 'class' parameter. It takes heap class object (hence J9Object * type is used), but not (pointer  to) J9Class struct, as all other barrier types dealing with class do.

Caller itself probably converts J9Class struct to class object prior to calling this API. If possible, I'd like to convert this API signature to be consistent with other. But if it's involves more changes in JIT/VM, we can do in it a separate change.

Reply:
Unfortunately, it's not so simple, it involves more changes at call sites in JIT/VM... for example:

```C++
void J9FASTCALL
fast_jitWriteBarrierClassStoreMetronome(J9VMThread *currentThread, j9object_t destinationClassObject, J9Object** destinationAddress, j9object_t objectBeingStored)
{
	currentThread->javaVM->memoryManagerFunctions->J9WriteBarrierPreClass(currentThread, destinationClassObject, destinationAddress, objectBeingStored);
}
```

Reply:
Which also brings up another point, the call site APIs should be updated as well like `fast_jitWriteBarrierClassStoreMetronome`

Reply:
in interim (or if we don't end up changing signature, at all), you can call the parameter dstClassObject (like JIT API does), so it's at least clear what object is representing the class.

Reply:
Done, I've also changed apis param names  in `ObjectAccessBarrierAPI.hpp` to use `dstClassObject`:

- preStaticStoreObject
- internalStaticPreStoreObject
- internalStaticPreStoreObjectSATB

They all chain together to call `J9WriteBarrierPreClass`

Reply:
Should these API names be changed in any way to mention "class"?

Reply:
>It's unfortunated that Pre variant is inconsistent in 'class' parameter. It takes heap class object (hence J9Object * type is used), but not (pointer to) J9Class struct, as all other barrier types dealing with class do.

>Caller itself probably converts J9Class struct to class object prior to calling this API. If possible, I'd like to convert this API signature to be consistent with other. But if it's involves more changes in JIT/VM, we can do in it a separate change.

We can probably start by making the changes in this file, by directly passing in clazz to top level apis (preStaticStoreObject  & internalStaticPreStoreObject) instead of the classobj... and do the conversion in  `internalStaticPreStoreObjectSATB` before calling `PreClass`

Reply:
> 
> 
> Should these API names be changed in any way to mention "class"?

It's a mixed bag of static vs class. Not sure atm what the prefered way is, so I'd leave it as-is for now.

Reply:
> We can probably start by making the changes in this file, by directly passing in clazz to top level apis (preStaticStoreObject & internalStaticPreStoreObject) instead of the classobj... and do the conversion in `internalStaticPreStoreObjectSATB` before calling `PreClass`

Let's do it a set of separate changes.